docs/protocol/extensions/string-dtypes/v1.0.rst

```diff
@@ -0,0 +1,136 @@
+===================================
+ String data types (version 1.0)
+===================================
+-----------------------------
+ Editor's draft 2 March 2022
+-----------------------------
+
+Specification URI:
+    http://purl.org/zarr/spec/protocol/extensions/string-dtypes/1.0
+Issue tracking:
+    `GitHub issues <https://github.com/zarr-developers/zarr-specs/labels/string-dtypes-v1.0>`_
+Suggest an edit for this spec:
+    `GitHub editor <https://github.com/zarr-developers/zarr-specs/blob/core-protocol-v3.0-dev/docs/protocol/extension/string-dtypes/v1.0.rst>`_
+
+Copyright 2022 `Zarr core development
+team <https://github.com/orgs/zarr-developers/teams/core-devs>`_ (@@TODO
+list institutions?). This work is licensed under a `Creative Commons
+Attribution 3.0 Unported
+License <https://creativecommons.org/licenses/by/3.0/>`_.
+
+----
+
+
+Abstract
+========
+
+This specification is a Zarr protocol extension defining data types
+for strings. It is an early draft and currently just describes existing support
+for NumPy string types that have already worked with zarr-python, but are not
+part of the core v3 spec.
+
+
+Status of this document
+=======================
+
+This document is a **Work in Progress**. It may be updated, replaced
+or obsoleted by other documents at any time. It is inappapropriate to
+cite this document as other than work in progress.
+
+Comments, questions or contributions to this document are very
+welcome. Comments and questions should be raised via `GitHub issues
+<https://github.com/zarr-developers/zarr-specs/labels/string-dtypes-v1.0>`_. When
+raising an issue, please add the label "string-dtypes-v1.0".
+
+This document was produced by the `Zarr core development team
+<https://github.com/orgs/zarr-developers/teams/core-devs>`_.
+
+
+Document conventions
+====================
+
+Conformance requirements are expressed with a combination of
+descriptive assertions and [RFC2119]_ terminology. The key words
+"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in the normative
+parts of this document are to be interpreted as described in
+[RFC2119]_. However, for readability, these words do not appear in all
+uppercase letters in this specification.
+
+All of the text of this specification is normative except sections
+explicitly marked as non-normative, examples, and notes. Examples in
+this specification are introduced with the words "for example".
+
+
+Extension data types
+====================
+
+Two extension data types are defined to represent zero-terminated bytestrings as
+well as fixed-length 32-bit unicode arrays.
+
+Fixed length byte strings (zero-terminated)
+-------------------------------------------
+
+These are fixed width strings corresponding to NumPy dtypes with `kind` 'S'.
+For backward compatibility with Python 2's ``str`` these are zero-terminated
```

Comment:
Zero-terminated strings normally refers to the C null-terminated string convention where the length is not explicitly indicated and instead the end of the string is indicated by a `'\0'` value.  I would call this zero-padded rather than zero-terminated, since there is no zero terminator if the string is exactly equal to the specified length.

Reply:
Yeah, I was a bit confused regarding this as the [NumPy dtype docs](https://numpy.org/doc/stable/reference/arrays.dtypes.html) refer to this type as "zero-terminated bytes".